plugins/discovery-gce/src/main/java/org/elasticsearch/cloud/gce/GceComputeServiceImpl.java

```diff
@@ -168,17 +127,17 @@ public GenericUrl run() {
     private JsonFactory gceJsonFactory;
 
     private final boolean validateCerts;
+
     @Inject
-    public GceComputeServiceImpl(Settings settings, NetworkService networkService) {
+    public GceComputeServiceImpl(Settings settings) {
         super(settings);
         this.project = PROJECT_SETTING.get(settings);
         this.zones = ZONE_SETTING.get(settings);
         this.gceHost = GCE_HOST.get(settings);
         this.metaDataUrl =  gceHost + "/computeMetadata/v1/instance";
         this.gceRootUrl = GCE_ROOT_URL.get(settings);
-        tokenServerEncodedUrl = metaDataUrl + "/service-accounts/default/token";
+        this.tokenServerEncodedUrl = metaDataUrl + "/service-accounts/default/token";
```

Comment:
why do we need all these urls if we have a dedicated service to fetch meta data? sounds duplicate. Also GceComputeService is a duplicate - the compute is already part of the Gce prefix. Since this is only in charge of resolving  instances of the same project, maybe call it GceProjectService or GceInstancesService?